sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -1054,6 +1054,7 @@ public static Sink sink(Schema schema) {
     return new AutoValue_ParquetIO_Sink.Builder()
         .setJsonSchema(schema.toString())
         .setCompressionCodec(CompressionCodecName.SNAPPY)
+        .setRowGroupSize(0)
```

Comment:
Would it make sense to set a default size here instead of just `0`?

Reply:
There is already a default in `ParquetWriter.Builder` for `rowGroupSize` ([here](https://github.com/apache/parquet-mr/blob/0efbfaa9a7b02bbd2a5912313ca77a9550da8ea1/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java#L357)) if it is not explicitly set in the builder. So I decided to interpret 0 as fallback to that default (see line 1128 below). Do you feel that we need to introduce another default here?

Reply:
Yes, I just asked about this because imho `0` looks like a `magic number` here which makes it a bit confusing why it's set to 0. I think it would be more evident to set it to `ParquetWriter.DEFAULT_BLOCK_SIZE` by default. Wdyt?

Reply:
I agree with your point that `0` looks like magic here (how about adding a one line comment?). But the problem with picking a default here is that I feel this is something that needs to be left to the underlying writer. For example, I am not sure if `ParquetWriter.DEFAULT_BLOCK_SIZE` is even the "right" choice because there is also  [`ParquetWriter.DEFAULT_PAGE_SIZE`](https://github.com/apache/parquet-mr/blob/0efbfaa9a7b02bbd2a5912313ca77a9550da8ea1/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java#L43) which actually has a different value [here](https://github.com/apache/parquet-mr/blob/616c3d681c5506ebad83a7d3724e85da2154eac9/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L47) (i.e., 1MB vs 128MB for default *block* size).

It does seem to me that in some places `rowGroupSize` and `blockSize` are used interchangeably in `ParquetWriter`. For example [this constructor call](https://github.com/apache/parquet-mr/blob/0efbfaa9a7b02bbd2a5912313ca77a9550da8ea1/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java#L229) is passing its `blockSize` parameter as the `rowGroupSize` of the called constructor. If you feel strongly about taking a stance here, it is okay with me but because of these complexities in `ParquetWriter` I thought I just leave the choice of the default to that. WDYT?

Reply:
Yes, I see your point. So let's keep it as it is (or `-1` maybe?) just, please, add a comment on this why we set it to 0 here. Thanks!

Reply:
I thought a little more about this and decided to go with your original suggestion. Now I think it is actually not a bad idea to expose a little bit of complexities inside `ParquetWriter` here to give a signal to the user that `rowGroupSize` is actually used for block-size setting too (and there is a comment too, so that should be fine).

PTAL.

Reply:
Forgot to mention that another reason I went this route is that we don't need the `rowGroupSize==0` check in `open()` below (line 1128). If anywhere else we used `Sink.Builder` we should have accounted for this special case but with using the new default, we don't need such checks.